pkg/meta/types.go

```diff
@@ -15,6 +15,30 @@
 
 package meta
 
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+)
+
+const (
+	jsonIndent    = "  "
+	jsonWriteSize = 64 << 10
+)
+
+type bufWriter struct {
+	bufio.Writer
+}
+
+func (bw *bufWriter) write(s string) {
```

Comment:
@davies used a wrapper with panic to simplify code, any opposition to this?

Reply:
have this in `writeJSON` will simpler:
```
write = func(s string) {
        if _, err := bw.WriteString(s); err != nil {
		panic(err)
	}
}
```